Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -221,149 +262,258 @@ public List<InteractableEvent> InteractableEvents
+            set { SetState(InteractableStates.InteractableStateEnum.Collision, value); }
+        }
 
         /// <summary>
-        /// A voice command has occurred, this does not automatically reset
-        /// Can be reset using the SetVoiceCommand(bool) method.
+        /// A voice command has just occurred
         /// </summary>
-        public bool HasVoiceCommand { get; private set; }
+        public virtual bool HasVoiceCommand
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.VoiceCommand) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.VoiceCommand, value); }
+        }
 
         /// <summary>
         /// A near interaction touchable is actively being touched
         /// </summary>
-        public bool HasPhysicalTouch { get; private set; }
+        public virtual bool HasPhysicalTouch
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.PhysicalTouch) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.PhysicalTouch, value); }
+        }
 
         /// <summary>
-        /// Misc - custom: not set by Interactable
+        /// State that corresponds to miscellaneous/custom use by consumers
+        /// Currently not controlled by Interactable directly
         /// </summary>
-        public bool HasCustom { get; private set; }
+        public virtual bool HasCustom
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Custom) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Custom, value); }
+        }
 
         /// <summary>
-        /// A near interaction grabbable is actively being grabbed/
+        /// A near interaction grabbable is actively being grabbed
         /// </summary>
-        public bool HasGrab { get; private set; }
-
-        // internal cached states
-        protected State lastState;
-        protected bool wasDisabled = false;
+        public virtual bool HasGrab
+        {
+            get { return GetStateValue(InteractableStates.InteractableStateEnum.Grab) > 0; }
+            set { SetState(InteractableStates.InteractableStateEnum.Grab, value); }
+        }
 
         #endregion
 
-        // check for isGlobal or RequiresFocus changes
-        protected bool requiresFocusValueCheck;
-        protected bool isGlobalValueCheck;
+        protected State lastState;
 
-        // cache of current dimension
-        [SerializeField]
-        protected int dimensionIndex = 0;
+        // directly manipulate a theme value, skip blending
+        protected bool forceUpdate = false;
 
         // allows for switching colliders without firing a lose focus immediately
         // for advanced controls like drop-downs
         protected float rollOffTime = 0.25f;
         protected float rollOffTimer = 0.25f;
 
-        // cache voice commands
         protected string[] voiceCommands;
 
```

Comment:
doesn't do anything useful in the script

Reply:
Great catch, thanks